autometa/common/external/samtools.py

```diff
@@ -81,11 +43,33 @@ def sort(sam, bam, nproc=mp.cpu_count()):
         </path/to/output/alignment.bam>
     nproc : int, optional
         Number of processors to be used. By default uses all the processors of the system 
-    """
     
-    cmd = f'samtools view -@{nproc} -bS {sam} | samtools sort -@{nproc} -o {bam}'
-    run(cmd,bam)
+    Raises
+    ------
+    ChildProcessError
+        Samtools did not run successfully, returns retcode.
+    
+    """
 
+    cmd = f'samtools view -@{nproc} -bS {sam} | samtools sort -@{nproc} -o {bam}'
+    logger.debug(f'cmd: {cmd}')
+    samtools_out_dir = os.path.dirname(bam)
```

Comment:
If the \`bam\` provided is not a full path then the dirname could be empty. You could account for this with

`samtools_out_dir = os.path.dirname(os.path.abspath(bam))`

Reply:
Agree